### ISSUE TITLE
False positive error CS9503 "Required member cannot be less visible than the containing type" for member in nested type

### ISSUE THREAD

**Issue by sirduke** [dotnet/roslyn#61527]
**Version Used**: 
features/required-members

**Steps to Reproduce**:

```
internal class InternalOuterCsass
{
    public  class PublicInnerClass
    {
        internal required int field;
    }
}
```

**Expected Behavior**:
no error because effective accessibility of containing type is "internal"

**Actual Behavior**:
> error CS9503: Required member 'InternalOuterCsass.PublicInnerClass.field' cannot be less visible or have a setter less visible than the containing type

**Comment by jcouv** [comment]
I'm failing to see what the error protects in this scenario. We might consider tweaking the rules a bit.

**Comment by 333fred** [comment]
IMO, this is pretty analogous to how interface methods must be public in such a scenario, even though `PublicInnerClass` is effectively internal.

**Comment by 333fred** [comment]
It turned out to be easier to fix #61528 by specifying in terms of effective visibility, which changes the behavior in this bug as requested.
